cli/src/deploy.js

```diff
@@ -52,44 +34,46 @@ export async function deployForDebug() {
 
   const imageTag = `${dockerhubUsername}/${iDappName}:${idappVersion}`;
 
-  await checkDockerDaemon();
-
   try {
-    await dockerBuild({
+    // just start the spinner, no need to persist success in terminal
+    spinner.start('Checking docker daemon is running...');
+    await checkDockerDaemon();
+
+    spinner.start('Building docker image...\n');
+    const buildLogs = [];
+    const imageId = await dockerBuild({
       tag: imageTag,
+      progressCallback: (msg) => {
+        buildLogs.push(msg); // do we want to show build logs after build is successful?
+        spinner.text = spinner.text + msg;
+      },
     });
-    await pushDockerImage({ tag: imageTag });
-  } catch (e) {
-    console.log(
-      chalk.red(
-        `\n An error occurred during the deployment of the non-tee image: ${e.message}`
-      )
-    );
-    return;
-  }
+    spinner.succeed(`Docker image built (${imageId}) and tagged ${imageTag}`);
 
-  try {
-    // Sconifying iDapp
-    const sconifySpinner = ora(
-      'Sconifying your iDapp, this may take a few minutes ...'
-    ).start();
+    spinner.start('Pushing docker image...\n');
+    await pushDockerImage({
+      tag: imageTag,
+      dockerhubAccessToken,
+      dockerhubUsername,
+      progressCallback: (msg) => {
+        spinner.text = spinner.text + msg;
+      },
+    });
+    spinner.succeed(`Pushed image ${imageTag} on dockerhub`);
+
+    spinner.start(
+      'Transforming your image into a TEE image and deploying on iExec, this may take a few minutes ...'
+    );
     const { dockerHubUrl } = await sconify({
       mainSpinner: sconifySpinner,
```

Comment:
`sconifySpinner` does not seem to exist anymore.

Maybe:
```suggestion
      mainSpinner: spinner,
```